### PR TITLE
Added CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by Learningcomputers777**
#27

### CODE REVIEW

**Comment by Learningcomputers777** [comment]
Added installation instructions for developers. As mentioned In the documentation link (https://surfactant--63.org.readthedocs.build/en/63/contribute.html#installation-for-developers).

**Comment by nightlark** [comment]
Thanks for the PR! It looks like someone else also opened a PR yesterday adding a contributing file, though it also seems to be just a direct copy of a template and needs a number of changes. I would like to give the other person a day or so to see if they make requested changes.

Otherwise, my thinking is this one looks like it covers the essentials and has been tailored to this project — maybe some info could be added to the section on submitting a bug report to mention https://github.com/LLNL/Surfactant/security for reporting security issues/vulnerabilities, and a link to the code of conduct, then it will be ready to merge?

**Comment by Learningcomputers777** [comment]
I would work on that and submit another other PR.

**Comment by Learningcomputers777** [comment]
I have added the CODE OF CONDUCT link of contributor covenant which has widely used and adopted by many firms.
(https://www.contributor-covenant.org).
Also mentioned the link to report the security issues and vulnerabilities in the CONTRIBUTING.md file.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+1. If you find a bug or have a feature request, please [open a new issue](https://github.com/LLNL/Surfactant/issues) and provide detailed information about the problem.
+2. If you find security issues or vulnerabilities, please [report here](https://github.com/LLNL/Surfactant/security)
+
+## Making Contributions
+
+We welcome contributions from the community. To contribute to this project, follow these steps:
+
+1. Fork the repository on GitHub.
+2. Clone your forked repository to your local machine.
+
```
Comment (nightlark) [inline CONTRIBUTING.md:16]:
```suggestion

All contributions to Surfactant are made under the MIT license (MIT).

```

**Review by nightlark — COMMENTED** [review]
Great -- just looked over it again, a few more small changes and this will be ready to merge.

**Comment by Learningcomputers777** [comment]
Can you make it Hackoctober fest label?

**Comment by Learningcomputers777** [comment]
I would work on all those and submit another PR.
Can you make it Hackoctober label?

**Comment by nightlark** [comment]
It should be possible to go to https://github.com/LLNL/Surfactant/pull/63/files and click "add suggestion to batch" to commit them as a group -- probably the easiest way to make the changes.

The repository has the `hacktoberfest` label, so when the PR is approved/merged you'll get credit (may take a little bit for the hacktoberfest website to update its status).

**Comment by nightlark** [comment]
If needed yes -- from their FAQ (and previous years experience with Hacktoberfest) any of the three conditions are sufficient for it to count ("Your PR/MRs must be merged, have the "hacktoberfest-accepted", or have an overall approving review.`)

**Comment by Learningcomputers777** [comment]
Done with the changes.Can you review it and merge it, if there are any more changes to be made I would work on that.

**Review by nightlark — APPROVED** [review]

**Comment by nightlark** [comment]
Regarding other changes, currently the CODE_OF_CONDUCT.md file we have in the repository (https://github.com/LLNL/Surfactant/blob/main/.github/CODE_OF_CONDUCT.md) is for version 2.0 of the contributor covenant -- that could get updated to the 2.1 version.

**Comment by Learningcomputers777** [comment]
Ok I will create an issue and will make the changes.I guess there is no need for hackoctober  label because yesterday for my pull request which is merged it was shown in my profile in hackoctober.

**Pull request merged**